### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by oovillagran**
<h1>Hi there 👋</h1>

In this banch, I implemente [Redux Toolkit](https://redux-toolkit.js.org/) inside of bookstore application.  Following the requirements:

**Project Requirements**

- Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Create a directory that will contain all your Redux logic (/src/redux).
- Configure a Redux store (/src/redux/store.js)
- Define a ```slice``` of state for books that:
     - Includes an array of books (initial state: empty array).
     - Includes a reducer that adds a book.
     - Includes a reducer that removes a book.
- Define a ```slice``` of state for categories that:
     - Includes an array of categories (initial state: empty array)
     - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string).

**NOTE:** *No changes in React components are required in this project.*

<h2>Thanks for your kind feedback!</h2>

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      const newBook = action.payload;
+      state.booksList.push(newBook);
+    },
+    removeBook: (state, action) => {
+      state.booksList.filter((book) => book.id !== action.payload);
+    },
+  },
+});
+
+export const { removeBook } = bookSlice.actions;
```
Comment (pinkmoon25) [inline src/redux/books/bookSlice.js]: - [x] Great work so far, I think we should also export the addBook action as well as we are going to need it to dispatch actions to reducers

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+  status: 'under construction',
+};
```
Comment (pinkmoon25) [inline src/redux/categories/categoriesSlice.js]: - [ ] According to the requirements :point_down: the initial state should be just an empty array
image

something like : 
```
initialState: {
    categories: [],
  }
```
then you can check and return like 
```
 action.payload === "Under construction" ? "Under construction" : state.categories
```

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @oovillagran , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linter checks :heavy_check_mark: 
github flow :heavy_check_mark: 
ducks pattern :heavy_check_mark: 
redux toolkit :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/bookSlice.js (hunk 1 above)
Comment (oovillagran) [reply src/redux/books/bookSlice.js]: Done.

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (oovillagran) [reply src/redux/categories/categoriesSlice.js]: Done!

Comment (oovillagran) [reply src/redux/categories/categoriesSlice.js]: Thanks @PinkMoon25 !

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @oovillagran👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oovillagran** [comment]
Thanks @thecodechaser !!

**Pull request merged**